### DISCUSSION TITLE
Is it possible to independently set the CPU or GPU device in the program's startup parameters?

### DISCUSSION THREAD

**Discussion by gtygo** [ggml-org/llama.cpp#10460] [Q&A]
Hi, I would like to use either the CPU or GPU during runtime. Currently, it seems that this can only be controlled at the compilation stage using options like GGML_METAL or GGML_BLAS. Is there a way to configure the code to select my computation device before initializing ggml_backend?

**Comment by ExtReMLapin** [comment]
For GPU, just set tensor split to only one of the GPU and split mode to None, it will remove the overhead and reach same performances of CUDA_VISIBLE_DEVICES
